### PR TITLE
Added methods for easier construction of Paths

### PR DESCRIPTION
**Pull request by Nohus**
If you already have a list of points it doesn't feel great to have to add them to `PathDataBuilder` one by one, so I added a `addPoints` method to add a list of points. Also made the methods chainable, and added a `makePathData()` method to completely avoid having to use `PathDataBuilder` manually for simple cases.

### CODE REVIEW

**Comment by Nohus** [comment]
Also added `removeAllPaths()`, which seemed missing among the existing `removeAllMarkers()` and `removeAllLayers()`.

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/paths/PathComposer.kt [hunk 1]
```diff
@@ -76,12 +76,19 @@ class PathDataBuilder internal constructor(
     /**
      * Add a point to the path. Values are relative coordinates (in range [0f..1f]).
      */
-    fun addPoint(x: Double, y: Double) {
-        points.add(
-            Offset((x * fullWidth).toFloat(), (y * fullHeight).toFloat())
-        )
+    fun addPoint(x: Double, y: Double) = apply {
```
Comment (p-lr) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/paths/PathComposer.kt:79]: is `apply` necessary here?

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/paths/PathComposer.kt [hunk 2]
```diff
@@ -76,12 +76,19 @@ class PathDataBuilder internal constructor(
...
+        points.add(createOffset(x, y))
     }
 
+    /**
+     * Add points to the path. Values are relative coordinates (in range [0f..1f]).
+     */
+    fun addPoints(points: List<Pair<Double, Double>>) = apply {
```
Comment (p-lr) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/paths/PathComposer.kt:86]: Same here, unless i'm missing something

Comment (p-lr) [reply mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/paths/PathComposer.kt:86]: oh, then ok.

**Comment by p-lr** [comment]
+1 for merge

**Comment by p-lr** [comment]
Then I also realized that the overload must return the created `PathData`, as it's necessary for the `updatePath` api.

**Comment by Nohus** [comment]
I've checked it out, but I'm not sure I understand the advantage adding the points separately. How does it "defeat the point of creating a sequence"?

**Comment by p-lr** [comment]
Yes I wasn't clear. The sequence is there to read the lines of the file and add points to the path line by line, instead of reading all lines, keep then in memory, and add points from them.

**Comment by p-lr** [comment]
Imagine you have a 10Gb file to read to make a (huge) path. This is ridiculous but imagine that. Since the points of the path are going to take a non negligible amount of memory, what we want to do is allocate just enough memory (construct the path on the go). What we don't want to do is a double taxation on memory (e.g store all points in a temporary data structure, then build the path from it).

**Comment by Nohus** [comment]
I see what you mean now, that makes sense, but depends on the use case. In my case I already have a `List` of points so there is no value in trying to make a sequence out of it, they are already in memory.

**Comment by p-lr** [comment]
If the list of points are already in memory _as a list of Pair of doubles_, the following syntax can be used:

```kotlin
addPath(trackName, color = color, width = 12.dp) {
   addPoints(pointList)
}
```

Not as neat as `addPath(trackName, pointList, color = color, width = 12.dp)` but still better than before.

However, if the list of points is not stored as a list of Pair of doubles, then using the above syntax, `pointList` has to be converted, with a double taxation on memory. Depending on the size of the path you might not care at all. But if you care about memory, making a sequence from the existing point list makes sense (and use `addPoint` instead of `addPoints`).

**Pull request merged**